pkg/ccl/changefeedccl/changefeed_processors.go

```diff
@@ -1058,7 +1058,8 @@ func (j *jobState) checkpointCompleted(ctx context.Context, checkpointDuration t
 
 	j.metrics.CheckpointHistNanos.RecordValue(checkpointDuration.Nanoseconds())
 	j.lastProgressUpdate = j.ts.Now()
-	j.checkpointDuration = time.Duration(j.metrics.CheckpointHistNanos.Mean())
+	j.checkpointDuration = time.Duration(j.metrics.CheckpointHistNanos.Mean(
+		j.metrics.CheckpointHistNanos.ToPrometheusMetric()))
```

Comment:
I think most of the savings are here: https://github.com/cockroachdb/cockroach/pull/116581/files#diff-8453981e212eb3eee4afbfde1f551dc981960870b369580d9b4f911b9108533cR704

We now call ToPrometheusMetricWindowed only once for the loop rather than `len(metric.RecordHistogramQuantiles)` times.

Reply:
ack.

Reply:
Correct, key word I guess is the allocations are reduced when **recording** histogram metrics (as in, recording to TSDB). Agreed that the # of allocations in a use case like what we're doing here with the `CheckpointHistNanos` has the same number of allocations. 

I hear you re: ergonomics of the call sites, though. I have a patch coming that will help clean that up.

Reply:
Do we have to allocate every time we call ToPrometheusMetrics?

Reply:
We do. We could use a `sync.Pool`, but the heavy allocations don't come from the initial construction of the objects. Instead, they come within 3rd party Prometheus code that "fills in" that allocated struct via a call to `Write()`: https://github.com/cockroachdb/cockroach/blob/master/pkg/util/metric/metric.go#L436

So, introducing a `sync.Pool` wouldn't have a significant impact on the problem we're trying to solve for this release blocker.

Reply:
I see... I thought this code changed things like recording a metric to require toPrometheusmetric call (that would be bad)... It didn't.  So, yeah, I agree -- sync.pool is not going to help.